### PR TITLE
docs: add migration guide for typography changes

### PR DESCRIPTION
**Pull request by snitin315**

### CODE REVIEW

File: Migration-v11.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+## Manual Migration Guide
+
+### Typography Components
+
+- **The `Title` component has been removed in favor of the `Heading` component.**
+
+  ```diff
+  - <Title size="xlarge">hello world</Title>
+  + <Heading size="2xlarge">hello world</Heading>
```
Comment (snitin315) [inline Migration-v11.md:42]: We can host it on the storybook, initially, I tried the same but the `diff` blocks aren't getting highlighted like markdown.

File: Migration-v11.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+### 🚧 Watch Out for Limitations & Edge Cases
+
+> [!IMPORTANT]
+>
+> There might be some situations where the codemod falls short. If you encounter errors, handle those cases manually for a seamless transition. Happy coding!
+
+#### Typography
+
+1. The codemod doesn't handle the migration of conditionally rendered props. Take a moment to manually inspect and update such cases.
```
Comment (saurabhdaware) [inline Migration-v11.md:27]: According the test here, it does work on conditionally rendered props right https://github.com/razorpay/blade/pull/1846/files

File: Migration-v11.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  + <Heading size="xlarge">hello world</Heading>
+
+  - <Title size="medium">hello world</Title>
+  + <Heading size="xlarge">hello world</Heading>
+
+  - <Title size="small">hello world</Title>
+  + <Heading size="large">hello world</Heading>
+  ```
+
+- **The `type` prop has been removed from the `Text`, `Heading`, & `Display` components.**
```
Comment (saurabhdaware) [inline Migration-v11.md:54]: This will have a migration once we are done with color changes. `type` and `contrast` is removed and can be set from `color` prop.

We'll add this to color changes and not typography changes

File: Migration-v11.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  + <Heading>hello world</Heading>
+
+  - <Display type="normal/subtle/muted/placeholder/subdued">hello world</Display>
+  + <Display>hello world</Display>
+  ```
+
+- **The `variant` prop has been removed from the `Heading` component.**
+
+  ```diff
+  - <Heading variant="regular/subheading">hello world</Heading>
```
Comment (saurabhdaware) [inline Migration-v11.md:70]: `variant="subheading"` migrates to Text small, `variant="regular"` is removed

File: Migration-v11.md (hunk 2 above)
Comment (snitin315) [reply Migration-v11.md:27]: So `Title` gets changed to `Heading` but the `size` prop remains as it is.

File: Migration-v11.md (hunk 4 above)
Comment (snitin315) [reply Migration-v11.md:70]: Oh! Need to handle this in the codemod as well.

File: Migration-v11.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
+# Migration Guide for v11 (Brand Refresh)
+
+## Migration with Codemod
+
+**Step1:** Install this version of `@razorpay/blade`.
```
Comment (kamleshchandnani) [inline Migration-v11.md:5]:
```suggestion
**Step1:** Install this version of Blade as `yarn add @razorpay/blade:rebranding@v<<this version>>`.
```

File: Migration-v11.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+**Step2:** Execute the codemod using the following command:
```
Comment (kamleshchandnani) [inline Migration-v11.md:7]: mention that migration should be done page by page.
also add that till the whole project is migrated they'll need to have 2 blade packages installed and that we can't use 2 versions of blade on the same page. add blade example to showcase that

File: Migration-v11.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+> [!WARNING]
+>
+> While this codemod covers most cases, it's always good to double-check and test your code to catch any missed nuances.
+
+### 🚧 Watch Out for Limitations & Edge Cases
+
+> [!IMPORTANT]
+>
+> There might be some situations where the codemod falls short. If you encounter errors, handle those cases manually for a seamless transition. Happy coding!
+
```
Comment (kamleshchandnani) [inline Migration-v11.md:24]: mention that there will be cases where the script will add that weird string and people shall talk to designs and fix those manually

File: Migration-v11.md (hunk 5 above)
Comment (snitin315) [reply Migration-v11.md]: Updated

File: Migration-v11.md (hunk 6 above)
Comment (snitin315) [reply Migration-v11.md]: Added

File: Migration-v11.md (hunk 7 above)
Comment (snitin315) [reply Migration-v11.md:60]: added

File: Migration-v11.md [hunk 8]
```diff
@@ -0,0 +1,371 @@
...
+      </div>
+    </BladeProvider>
+  );
+};
+```
+
+```jsx
+// OldPage.tsx (v10)
+import { BladeProvider } from '@razorpay/blade'; // <-- v10
+import { payment } from '@razorpay/blade/tokens';
```
Comment (saurabhdaware) [inline Migration-v11.md:30]:
```suggestion
import { paymentTheme } from '@razorpay/blade/tokens';
```

File: Migration-v11.md
Comment (kamleshchandnani) [inline Migration-v11.md:1]: @snitin315 lets remove this file since we have moved it to the handbook?

Comment (snitin315) [reply Migration-v11.md:1]: @kamleshchandnani Done.

**Review by kamleshchandnani — APPROVED** [review]
